lib/datasource.js

```diff
@@ -327,13 +327,22 @@ DataSource.prototype.setup = function (name, settings) {
 
     }.bind(this);
 
-    if ('function' === typeof connector.initialize) {
-      // Call the async initialize method
-      connector.initialize(this, postInit);
-    } else if ('function' === typeof connector) {
-      // Use the connector constructor directly
-      this.connector = new connector(this.settings);
-      postInit();
+    try {
+      if ('function' === typeof connector.initialize) {
+        // Call the async initialize method
+        connector.initialize(this, postInit);
```

Comment:
Makes sense to refactor this. Created waffle task https://github.com/strongloop-internal/scrum-loopback/issues/797 to track this.